src/main/java/com/google/devtools/build/lib/remote/common/RemoteCacheClient.java

```diff
@@ -63,6 +65,26 @@ public int hashCode() {
     }
   }
 
+  @AutoValue
+  abstract class CachedActionResult {
+    public static CachedActionResult create(ActionResult actionResult, String cacheName) {
```

Comment:
Unless we are going to support more cache types, allowing caller to construct arbitrary cache name could lead to unintentional bugs.

Please consider only offering 4 builder methods: `remoteHit`, `diskHit`, `remoteMiss`, `diskMiss`.

Reply:
I don't think we need the misses, but removed create.